rerun_cpp/src/rerun/collection.hpp

```diff
@@ -60,16 +67,68 @@ namespace rerun {
         }
 
         /// Construct using a `CollectionAdapter` for the given input type.
-        template <typename T>
-        Collection(T&& input) : Collection(Adapter<T>()(std::forward<T>(input))) {}
+        template <
+            typename TContainer, //
+            // Avoid conflicting with the copy/move constructor.
+            // We could implement this also with an adapter, but this might confuse trait checks like `std::is_copy_constructible`.
+            typename = std::enable_if_t<
+                !std::is_same_v<std::remove_reference_t<TContainer>, Collection<TElement>>> //
+            >
+        Collection(TContainer&& input)
+            : Collection(Adapter<TContainer>()(std::forward<TContainer>(input))) {}
+
+        /// Copy constructor.
+        ///
+        /// If the data is owned, this will copy the data.
+        /// If the data is borrowed, this will copy the borrow,
+        /// meaning there's now (at least) two collections borrowing the same data.
+        Collection(const Collection<TElement>& other) : ownership(other.ownership) {
+            switch (other.ownership) {
+                case CollectionOwnership::Borrowed: {
+                    storage.borrowed = other.storage.borrowed;
+                    break;
+                }
+
+                case CollectionOwnership::VectorOwned: {
+                    storage.vector_owned = other.storage.vector_owned;
```

Comment:
⚠️ This assigns to an uninitialized vector! You need to use placement-new here

Reply:
nice catch!